homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -366,6 +367,8 @@ def device_id(self):
     @property
     def serial(self):
```

Comment:
Instead of changing this field, I think we should change the unique id field since `f"{self._bridge_serial}_{self.device_id}"` is not the serial number of the device

Reply:
Unfortunately, if we only change it there, it still fails in the same way.  Lots of entities get tied into the first device that's registered with no serial.  Somehow the device creation is failing (quietly) due to it returning 'None' in the serial property.

This is the result of only changing the unique_id:

![image](https://user-images.githubusercontent.com/24459240/187492173-d852e560-aeb0-4d82-a33b-9f0875d0cc73.png)

Reply:
I think the problem is right here:

https://github.com/home-assistant/core/blob/e48d493db495d27427a44647e50eaa4e0f0c1919/homeassistant/components/lutron_caseta/__init__.py#L345-L351

the identifier is using self.serial, which returns None

Reply:
Pull the unique id change into the base class and change the device entry to use it instead of the serial and this should be GTG

Reply:
ok, i'll do some testing.  Nervous to make changes that might affect all currently deployed caseta installations.

Reply:
Should be fine since you have the None check. The current hubs won't have a none serial for devices, only groups and we already use that logic

Reply:
.. and I have all 3 of the currently supported system types so I wouldn't worry too much as I won't merge it if it breaks mine production systems 🤣

Reply:
haha.. ok thanks.  I'll test, commit and re-request review once ready.

Reply:
So if I change the identifiers to use unique_id, it duplicates all the device entries for my caseta install, that doesn't have None serials.  I would have to add a 'None' serial check in the __init__ class too, and use self.serial for the identifier if serial != None to avoid the duplication.

This is my caseta install when I change the identifier to unique_id:
![image](https://user-images.githubusercontent.com/24459240/187497949-c3301826-5856-4fea-a9c5-b27814831e49.png)

Reply:
Are we sure we want to do this?  Adding 2 lines to the serial property seems cleaner.

Reply:
I'm not sure what change you made, but it should be functionally identical if its not None since it should still return `self._device["serial"]` which is what its returning now unless `serial` is overridden in a subclass

Reply:
it's changing this:

identifiers={(DOMAIN, self.serial)},
to
identifiers={(DOMAIN, self.unique_id)},

It affects existing caseta installations.  The devices were previously registered using serial as the identifier.

Reply:
I pushed I change that should work. Unfortunately I'm walking out the door to the airport so I can't validate it.  Feel free to revert it if it doesnt'